src/gui/clips/SampleClipView.cpp

```diff
@@ -183,16 +182,22 @@ void SampleClipView::mouseDoubleClickEvent( QMouseEvent * )
 {
 	QString af = SampleLoader::openAudioFile();
 
-	if ( af.isEmpty() ) {} //Don't do anything if no file is loaded
-	else if (af == m_clip->m_sample.sampleFile())
-	{	//Instead of reloading the existing file, just reset the size
-		int length = static_cast<int>(m_clip->m_sample.sampleSize() / Engine::framesPerTick());
-		m_clip->changeLength(length);
+	// Don't do anything if no file is loaded
+	if (af.isEmpty()) { return; }
+	
+	if (af == m_clip->m_sample.sampleFile())
+	{
+		// Don't reload the same file again. Just reset the size.
+		m_clip->changeLengthToSampleLength();
```

Comment:
Was this refactoring necessary?

Reply:
It's more or less an application of the [boyscout rule](https://deviq.com/principles/boy-scout-rule). The view contained logic/actions that's part of the core/"business" and therefore I moved it to where it belongs.

Reply:
Since its low hanging fruit, it's not a big deal at all, but I would add some details about it in the commit description (something along the lines of also move around `setModified` logic to the correct spot) since this PR is doing more than what it said it would on the tin.

Reply:
It's already mentioned in the commit message of commit 14ce3cc0.